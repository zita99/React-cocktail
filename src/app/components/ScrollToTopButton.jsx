import React from 'react';

const ScrollToTopButton = () => {

  const handleOnClickScrollUp = (event, scrollDuration) => {
    event.preventDefault();
    const scrollValue = -window.scrollY / (scrollDuration / 20);
    const scrollInterval = setInterval(() => {
          if ( window.scrollY != 0 ) {
            window.scrollBy( 0, scrollValue );
          }
          else clearInterval(scrollInterval);
        }, 15);
      }

  return <div className='scroll-to-top' onClick={ event => handleOnClickScrollUp(event, 1000) } >^</div>
};

module.exports = ScrollToTopButton;
