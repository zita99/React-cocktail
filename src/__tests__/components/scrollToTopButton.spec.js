import ScrollToTopButton from '../../app/components/ScrollToTopButton.jsx';

describe('ScrollToTopButton', () => {

  it('renders correctly ', () => {
    expect(ScrollToTopButton()).toMatchSnapshot();
  });

});
