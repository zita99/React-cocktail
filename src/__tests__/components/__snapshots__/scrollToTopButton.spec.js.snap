// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollToTopButton renders correctly  1`] = `
<div
  className="scroll-to-top"
  onClick={[Function]}
>
  ^
</div>
`;
